services/core/shared/core/compat/sdk_v5/accounts.js

```diff
@@ -432,7 +434,51 @@ const getMultisignatureGroups = async account => {
 	return multisignatureAccount;
 };
 
-const getMultisignatureMemberships = async () => []; // TODO
+const getMultisignatureMemberships = async account => {
+	const multisignatureMemberships = {};
+	const multisignatureDB = await getMultisignatureIndex();
+	const memberInfo = await multisignatureDB.find({ memberAddress: account.address });
+
+	if (Array.isArray(memberInfo) && memberInfo.length) {
+		multisignatureMemberships.memberships = [];
+		await BluebirdPromise.map(
+			memberInfo,
+			async member => {
+				const result = await getIndexedAccountInfo({ address: member.groupAddress });
+				multisignatureMemberships.memberships.push({
+					address: result && result.address ? result.address : undefined,
+					username: result && result.username ? result.username : undefined,
+					publicKey: result && result.publicKey ? result.publicKey : undefined,
+				});
+			},
+			{ concurrency: memberInfo.length },
+		);
+	}
+	return multisignatureMemberships;
+};
```

Comment:
Please consider renaming it to `membershipInfo`, since the resolved info contains and focuses on all the memberships for the queried account address. And, since the `db.find()` always ensures the return of an array in the response, the check seems redundant.

```suggestion
const getMultisignatureMemberships = async account => {
	const multisignatureMemberships = {memberships: []};
	const multisignatureDB = await getMultisignatureIndex();
	const membershipInfo = await multisignatureDB.find({ memberAddress: account.address });

	await BluebirdPromise.map(
		membershipInfo,
		async membership => {
			const result = await getIndexedAccountInfo({ address: membership.groupAddress });
			multisignatureMemberships.memberships.push({
				address: result && result.address ? result.address : undefined,
				username: result && result.username ? result.username : undefined,
				publicKey: result && result.publicKey ? result.publicKey : undefined,
			});
		},
		{ concurrency: membershipInfo.length },
	);
	return multisignatureMemberships;
};
```